src/utils/error-handle.js

```diff
@@ -0,0 +1,6 @@
+var util = require('util');
+
+module.exports = function handleSystemError(message, ...objs){
```

Comment:
There's a need for more debugging information, especially when a non VFocus is passed to one of these functions. Instead of littering the code base with `util.inspect` it makes more sense to keep this operation in one place, I hadn't considered adding a new Error object.

In terms of adding the errors themselves: there are allot of assumptions that we are operating on a vFocus throughout the code base (using `takeWhile` for example) and, to me, it makes more sense to be explicit about whats going on here.